### PR TITLE
Fix the problem of handling multi submitted input step incorrectly

### LINKED ISSUE #366: Multi submitted input step was not handled correctly
### What happened?
If we submit input step twice, we will got the following response from `devops-apiserver`:

```json
{"allow":"false","message":"<nil>"}
```

It is worth mentioning that there is no any related error log found.

I guess this process is handled at code below:

https://github.com/kubesphere/ks-devops/blob/c96f7386e7cc716e7d2d9269c7376cd56921f07a/pkg/kapis/devops/v1alpha2/devops.go#L453-L458

### Expected response
Actually, we expect the result as below:

```json
{
  "message" : "Error processing Input Submit request. This Run instance does not have an Input with an id of 'Deploy-to-prod'.",
  "code" : 400,
  "errors" : [ ]
}
```

See also https://github.com/kubesphere/ks-devops/issues/363.

/kind enhancement

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR dose

Return friendly error if current step is unapprovable.

### Why we need it

Please see #366 

### Which issue this PR fixes

#366 

### Steps to test

Please replace docker image of `devops-apiserver` deployment with image name below:

```bash
johnniang/devops-apiserver:dev-v3.2.0-43358c2
```

- Create a simple approvable Pipeline and run, which Jenkinsfile is below:

```groovy
pipeline {
  agent none
  stages {
    stage('Input test') {
      steps {
        input(id: 'deploy-to-prod', message: 'deploy to prod?')
      }
    }
  }
}
```

- Approve the PipelineRun and copy request `http://demo:30880/kapis/devops.kubesphere.io/v1alpha2/devops/test-projectl82l5/pipelines/approable-test/runs/33/nodes/4/steps/5/` as CURL(bash)

image

- Request the same request again, and you will get the response:

```bash
* upload completely sent off: 23 out of 23 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Vary: Accept-Encoding
< content-length: 61
< content-type: text/plain; charset=utf-8
< date: Wed, 24 Nov 2021 09:49:48 GMT
< connection: close
<
* Closing connection 0
{"allow":"false","message":"the step state was unapprovable"}%
```

/kind bug
/cc @kubesphere/sig-devops

### CODE REVIEW

File: pkg/kapis/devops/v1alpha2/devops.go [hunk 1]
```diff
@@ -422,10 +424,12 @@ func (h *ProjectPipelineHandler) hasSubmitPermission(req *restful.Request) (hasP
 			}
 
 			for _, step := range node.Steps {
-				if step.ID != stepId || step.Input == nil {
+				if step.ID != stepId {
 					continue
 				}
-
+				if step.Input == nil {
+					err = errors.New("the step state was unapprovable")
```
Comment (LinuxSuRen) [inline pkg/kapis/devops/v1alpha2/devops.go:431]: Suggestion:

`err = fmt.Errorf("the step '%s' is not approvable", step.DisplayName)`

Comment (JohnNiang) [reply pkg/kapis/devops/v1alpha2/devops.go]: Done. Please see https://github.com/kubesphere/ks-devops/pull/374/commits/10c0284ca2946f5ddf224c7a7b6ae8f07579ed35.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Pull request merged**